hexrd/ui/mask_manager_dialog.py

```diff
@@ -136,6 +137,22 @@ def toggle_threshold_visibility(self, status):
         if self.threshold_cb is not None:
             self.threshold_cb.setChecked(status)
 
+    def masks_changed(self):
+        if self.image_mode == ViewType.polar or self.image_mode == ViewType.stereo:
+            HexrdConfig().polar_masks_changed.emit()
+        elif self.image_mode == ViewType.raw:
+            HexrdConfig().raw_masks_changed.emit()
+
+    def update_export_options(self):
+        hidden = self.ui.hide_masks.isChecked()
+        visible = len(HexrdConfig().visible_masks)
+        enabled = (not hidden) and visible
+        self.ui.export_masks.setEnabled(enabled)
+        self.ui.panel_buffer.setEnabled(enabled)
+        tip = '' if enabled else 'Masks hidden. Only visible masks are used.'
+        self.ui.export_masks.setToolTip(tip)
+        self.ui.panel_buffer.setToolTip(tip)
```

Comment:
It looks to me that exporting the masks actually writes out all masks (both visible and invisible), and it also saves the visibilities. I think this is desired behavior.

However, that means that it doesn't make sense to disable the `export_masks` button depending on whether masks are visible or not. I think the only time we'd want to disable the `export_masks` button is if there are no masks at all.

Also, applying the masks to the panel buffer also uses all masks (visible and invisible). I don't know if this is the desired behavior (like, it might make more sense to only apply **visible** masks to the panel buffers), but unless we change that, we also shouldn't disable the panel buffer button based upon the visibilities of the masks. I think this one also should only be disabled if there are no masks at all.

But it is really great to have tooltips when buttons are disabled that explain why they are disabled!

Reply:
For now I have just removed the code that disables the export but I agree, I think it makes sense to only use visible masks. To keep things consistent I'll skip that logic in this PR and I can open another PR with the changes to disable export/add tooltips/only use visible masks.